### DISCUSSION TITLE
Using RTT read and write with PyOCD Api

### DISCUSSION THREAD

**Discussion by RehabAbdAllah** [pyocd/pyOCD#1707] [Q&A]
Hi ,
I am looking to integrate pyocd in an automation framework to read and write to device via rtt .
Can you please share how to use RTT with pyocd api for reading and writing logs ?
Thanks

**Comment by Teslafly** [comment]
It may be helpful to look at the rtt server block for rtt usage examples.
https://github.com/pyocd/pyOCD/blob/main/pyocd/utility/rtt_server.py

But I do agree that the python api page is pretty lacking and could really use some more fleshed out examples. The config file page too.
https://pyocd.io/docs/api_examples.html
